RestaurantReservation.API/Base/AppControllerBase.cs

```diff
@@ -0,0 +1,39 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using RestaurantReservation.Core.Bases;
+using System.Net;
+
+namespace RestaurantReservation.API.Base;
+
+[ApiController]
+public class AppControllerBase : ControllerBase
+{
+    private IMediator _mediatorInstance;
+    protected IMediator Mediator => _mediatorInstance ??= HttpContext.RequestServices.GetService<IMediator>();
```

Comment:
from were we get _mediatorInstance?

Reply:
It's initialized using HttpContext.RequestServices.GetService<IMediator>(),
 which retrieves an instance of IMediator from the current request's service